userspace/libpman/src/lifecycle.c

```diff
@@ -28,6 +29,43 @@ int pman_open_probe() {
 	return 0;
 }
 
+int pman_prepare_progs_before_loading() {
+	/*
+	 * Probe required features for each bpf program, as requested
+	 */
+	for(int ev = 0; ev < PPM_EVENT_MAX; ev++) {
+		int num_skipped = 0;
+		int idx = 0;
+		event_prog_t *progs = event_prog_table[ev];
+		for(; idx < MAX_FEATURE_CHECKS && progs[idx].name != NULL; idx++) {
+			if(progs[idx].feat > 0) {
+				if(libbpf_probe_bpf_helper(BPF_PROG_TYPE_RAW_TRACEPOINT, progs[idx].feat, NULL) ==
+				   0) {
+					// Required feature not present
+					struct bpf_program *p =
+					        bpf_object__find_program_by_name(g_state.skel->obj, progs[idx].name);
+					if(p) {
+						bpf_program__set_autoload(p, false);
```

Comment:
What happens to programs that are not the default one when they are not picked? Do they still get loaded into the kernel and not used? i.e: is `recvmmsg_old_x` still loaded into the kernel when we decide to use `recvmmsg_x`?

Reply:
> Do they still get loaded into the kernel and not used?

Exactly. I will make some changes to always disable all unused programs.